From: martin f. krafft <madduck@debian.org>
Subject: Disable udev incremental assembly

Upstream enabled incremental assembly in the udev rules by default for 3.1.3,
but this is too early given the pending release of squeeze. Hence, this patch
simply reverts the change.

Signed-off-by: martin f. krafft <madduck@debian.org>

---
 udev-md-raid.rules |   13 ++++++-------
 1 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/udev-md-raid.rules b/udev-md-raid.rules
index f9607f3..c9a4f0e 100644
--- a/udev-md-raid.rules
+++ b/udev-md-raid.rules
@@ -1,13 +1,13 @@
 # do not edit this file, it will be overwritten on update
 
 SUBSYSTEM!="block", GOTO="md_end"
-
-# handle potential components of arrays
-ENV{ID_FS_TYPE}=="linux_raid_member", ACTION=="remove", RUN+="/sbin/mdadm -If $name"
-ENV{ID_FS_TYPE}=="linux_raid_member", ACTION=="add", RUN+="/sbin/mdadm --incremental $env{DEVNAME}"
-
-# handle md arrays
 ACTION!="add|change", GOTO="md_end"
+ACTION=="change", GOTO="md_no_incr"
+
+# import data from a raid member and activate it
+#ENV{ID_FS_TYPE}=="linux_raid_member", IMPORT{program}="/sbin/mdadm --examine --export $tempnode", RUN+="/sbin/mdadm --incremental $env{DEVNAME}"
+# import data from a raid set
+LABEL="md_no_incr"
 KERNEL!="md*", GOTO="md_end"
 
 # partitions have no md/{array_state,metadata_version}, but should not
@@ -32,7 +32,6 @@ ENV{DEVTYPE}=="partition", ENV{MD_DEVNAME}=="*[0-9]", SYMLINK+="md/$env{MD_DEVNA
 
 IMPORT{program}="/sbin/blkid -o udev -p $tempnode"
 OPTIONS+="link_priority=100"
-OPTIONS+="watch"
 ENV{ID_FS_USAGE}=="filesystem|other|crypto", ENV{ID_FS_UUID_ENC}=="?*", SYMLINK+="disk/by-uuid/$env{ID_FS_UUID_ENC}"
 ENV{ID_FS_USAGE}=="filesystem|other", ENV{ID_FS_LABEL_ENC}=="?*", SYMLINK+="disk/by-label/$env{ID_FS_LABEL_ENC}"
 
-- 
tg: (deb573e..) debian/disable-incremental (depends on: master)
